packages/blade/src/components/Tooltip/Tooltip.native.tsx

```diff
@@ -0,0 +1,105 @@
+/* eslint-disable @typescript-eslint/restrict-plus-operands */
+import { arrow, shift, useFloating, flip, offset } from '@floating-ui/react-native';
+import React from 'react';
+import { Modal, TouchableOpacity } from 'react-native';
+import { TooltipArrow } from './TooltipArrowNative';
+import { TooltipContent } from './TooltipContent';
+import type { TooltipProps } from './types';
+import { ARROW_HEIGHT, ARROW_WIDTH } from './constants';
+import { getPlacementParts } from './utils';
+import { useTheme } from '~components/BladeProvider';
+
+const Tooltip = ({
+  content,
+  children,
+  placement = 'left',
+  onOpenChange,
+}: TooltipProps): React.ReactElement => {
+  const { theme } = useTheme();
+  const [isOpen, setIsOpen] = React.useState(false);
+
+  const gap = theme.spacing[2];
+  const [side] = getPlacementParts(placement);
+  const isHorizontal = side === 'left' || side === 'right';
+  const arrowRef = React.useRef();
+  const context = useFloating({
+    sameScrollView: false,
+    placement,
+    middleware: [
+      shift({ crossAxis: false, padding: gap }),
+      flip({ padding: gap }),
+      offset(gap + ARROW_HEIGHT),
+      arrow({
+        element: arrowRef,
+        padding: isHorizontal ? 0 : ARROW_WIDTH,
+      }),
+    ],
+  });
+
+  const { refs, floatingStyles } = context;
+
+  const handleOpen = React.useCallback(() => {
+    setIsOpen(true);
+    onOpenChange?.({ isOpen: true });
+  }, [onOpenChange]);
+
+  const handleClose = React.useCallback(() => {
+    setIsOpen(false);
+    onOpenChange?.({ isOpen: false });
+  }, [onOpenChange]);
+
+  // wait for animation to finish before unmounting modal
+  const [isVisible, setIsVisible] = React.useState(() => isOpen);
+  React.useEffect(() => {
+    const id = setTimeout(() => {
+      if (!isOpen) {
+        setIsVisible(false);
+      }
+    }, theme.motion.duration.gentle);
+
+    if (isOpen) {
+      setIsVisible(true);
+    }
+    return () => clearTimeout(id);
+  }, [isOpen]);
+
+  return (
+    <>
+      {/* Cloning the trigger children to enhance it with ref and event handler */}
+      {React.cloneElement(children, {
+        onTouchEnd: handleOpen,
```

Comment:
Can this override a passed `onTouchEnd` on the component?

Reply:
Good catch, yes it will. also happening on web side, added a mergeProps utility to merge the props/functions and then pass into the cloneElement